crates/relayer/src/channel.rs

```diff
@@ -300,14 +301,13 @@ impl<ChainA: ChainHandle, ChainB: ChainHandle> Channel<ChainA, ChainB> {
         chain: ChainA,
         counterparty_chain: ChainB,
         channel: WorkerChannelObject,
-        height: Height,
     ) -> Result<(Channel<ChainA, ChainB>, State), ChannelError> {
         let (a_channel, _) = chain
             .query_channel(
                 QueryChannelRequest {
                     port_id: channel.src_port_id.clone(),
                     channel_id: channel.src_channel_id.clone(),
-                    height: QueryHeight::Specific(height),
+                    height: QueryHeight::Latest,
```

Comment:
It seems that when getting a NewBlock we were looking at the state at the height of the NewBlock in order to determine the state of the channel, e.g. which step of the handshake it is at.
This doesn't seem right, I would expect the latest state to be the important one.
Can you confirm this @romac @ancazamfir ?